client/src/app/api/api/api.service.ts

```diff
@@ -115,4 +188,57 @@ export class ApiService {
       }
     );
   }
+
+  public post(endpoint: string, componentType: string, name: string, observe: any = 'body', reportProgress: boolean = false): any {
```

Comment:
please revisit the visibility modificators of your new methods and make sure if they actually need to be public.

Reply:
It needs to be public

Reply:
ok